nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -147,138 +193,200 @@ protected void init(final ProcessorInitializationContext context) {
         return properties;
     }
 
-    private int readLines(final InputStream in, final int maxNumLines, final OutputStream out, final boolean keepAllNewLines, final byte[] leadingNewLineBytes) throws IOException {
+    private int readLines(final InputStream in, final int maxNumLines, final long maxByteCount, final OutputStream out,
+                          final boolean includeLineDelimiter, final byte[] leadingNewLineBytes) throws IOException {
         final EndOfLineBuffer eolBuffer = new EndOfLineBuffer();
 
-        int numLines = 0;
         byte[] leadingBytes = leadingNewLineBytes;
+        int numLines = 0;
+        long totalBytes = 0L;
         for (int i = 0; i < maxNumLines; i++) {
-            final EndOfLineMarker eolMarker = locateEndOfLine(in, out, false, eolBuffer, leadingBytes);
+            final EndOfLineMarker eolMarker = countBytesToSplitPoint(in, out, totalBytes, maxByteCount, includeLineDelimiter, eolBuffer, leadingBytes);
+            final long bytes = eolMarker.getBytesConsumed();
             leadingBytes = eolMarker.getLeadingNewLineBytes();
 
-            if (keepAllNewLines && out != null) {
+            if (includeLineDelimiter && out != null) {
                 if (leadingBytes != null) {
                     out.write(leadingBytes);
                     leadingBytes = null;
                 }
-
                 eolBuffer.drainTo(out);
             }
-
-            if (eolBuffer.length() > 0 || eolMarker.getBytesConsumed() > 0L) {
-                numLines++;
+            totalBytes += bytes;
+            if (bytes <= 0) {
+                return numLines;
             }
-
-            if (eolMarker.isStreamEnded()) {
+            numLines++;
+            if (totalBytes >= maxByteCount) {
                 break;
             }
         }
-
         return numLines;
     }
 
-    private EndOfLineMarker locateEndOfLine(final InputStream in, final OutputStream out, final boolean includeLineDelimiter,
-        final EndOfLineBuffer eolBuffer, final byte[] leadingNewLineBytes) throws IOException {
-
+    private EndOfLineMarker countBytesToSplitPoint(final InputStream in, final OutputStream out, final long bytesReadSoFar, final long maxSize,
+                                                   final boolean includeLineDelimiter, final EndOfLineBuffer eolBuffer, final byte[] leadingNewLineBytes) throws IOException {
         int lastByte = -1;
         long bytesRead = 0L;
+        final ByteArrayOutputStream buffer;
+        if (out != null) {
+            buffer = new ByteArrayOutputStream();
+        } else {
+            buffer = null;
+        }
```

Comment:
I stared at this if-else for longer than I care to admit.  A ternary operator for conditional assignment to a final variable would have helped me, at least, maybe others too.